core/src/main/java/fr/sncf/osrd/envelope/MaxEnvelopeBuilder.java

```diff
@@ -0,0 +1,243 @@
+package fr.sncf.osrd.envelope;
+
+import static fr.sncf.osrd.envelope.MaxEnvelopeBuilder.EnvelopeChangeType.*;
+import static fr.sncf.osrd.envelope.MaxEnvelopeBuilder.EnvelopeChangeType.END;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class MaxEnvelopeBuilder {
```

Comment:
This file is way too close to MinEnvelopeBuilder, we can't maintain two files like that.

Is the only difference the use of comparison operators ? Maybe we could wrap them using our `CmpOperator`?

Reply:
We could delete `MinEnvelopeBuilder` and replace it by a `MRSPEnvelopeBuilder` which as a much simpler algorithm since `EnvelopePart` are flat.

Reply:
We implemented what @flomonster proposed.